frontend/components/SetupPage/Create/SetupEoaFunding.tsx

```diff
@@ -225,6 +234,12 @@ export const SetupEoaFunding = () => {
     goto(SetupScreen.SetupCreateSafe);
   }, 5000);
 
+  const eoaBalance = walletBalances.find((e) => e.isNative);
```

Comment:
should we check here chainId as well? like the `currentChain` or maybe even `homeChainId` from selectedAgentConfig?

+ probably can use useMasterBalances instead